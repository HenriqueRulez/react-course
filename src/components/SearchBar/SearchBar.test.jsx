import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SearchBar } from '.'

describe('<SearchBar />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<SearchBar handleChange={fn}  searchValue={'testando'}/>);
    const searchBox = screen.getByPlaceholderText(/type your search/i);

    expect(searchBox.value).toBe('testando');
  });

  it('should call handleChange on each key press', () => {
    const fn = jest.fn();
    render(<SearchBar handleChange={fn}  searchValue={'value to be searched'}/>);
    const searchBox = screen.getByPlaceholderText(/type your search/i);
    const searchParam = 'value to be searched';

    userEvent.type(searchBox, searchParam);
    expect(searchBox.value).toBe(searchParam);
    expect(fn).toHaveBeenCalledTimes(searchParam.length);
  });

  it('should match snapeshot', () => {
    const fn = jest.fn();
    const {container} = render(<SearchBar handleChange={fn}  searchValue={'value to be searched'}/>);

    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });
})