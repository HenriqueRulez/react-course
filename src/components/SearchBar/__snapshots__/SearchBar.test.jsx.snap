// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match snapeshot 1`] = `
<input
  placeholder="type your search"
  type="search"
  value="value to be searched"
/>
`;
